src/densities.jl

```diff
@@ -10,44 +10,39 @@ grid `basis`, where the individual k-points are occupied according to `occupatio
 It is possible to ask only for occupations higher than a certain level to be computed by
 using an optional `occupation_threshold`. By default all occupation numbers are considered.
 """
-@views @timing function compute_density(basis::PlaneWaveBasis{T}, ψ, occupation;
+function compute_density(basis::PlaneWaveBasis{T}, ψ, occupation;
+                         occupation_threshold=zero(T)) where {T}
+    Tρ = promote_type(T, real(eltype(ψ[1])))
+    compute_density(Tρ, basis, ψ, occupation; occupation_threshold)
```

Comment:
If we need this for perf, ok, but otherwise I'd prefer the old version without the extra layer for simplicity.